### PR TITLE
ClassMapper : Further removal of JSExposer.h

### PR DESCRIPTION
**Pull request by verpeteren**

### CODE REVIEW

**Comment by efyx** [comment]
The overall looks good to me however : 
- Moving JSExposer macros to JSMacros.h is a good thing, however I think we can remove all of NIDIUM_JS_SETTER, NIDIUM_JS_GETTER, JSPropertyAccessors, etc.. ?
- What rules are we following for using quote vs <> for includes ?

( ping @paraboul )

**Comment by verpeteren** [comment]
1) Indeed, this has not been merged yet. However the classmapper-av branch builds upon this.
2) We can remove the macro's as we go along; classmapper-av moves several macros out of JSExposer.h.
3) Rather simple: local/project includes get a '#include "xxx.h"', external libraries get a '#include <xxx.h>'
The order of the includes could be:
a) The corresponding h file if it a c/cpp file
b) system includes
c) library includes
d) local includes

File: src/Binding/ClassMapper.h [hunk 1]
```diff
@@ -758,7 +758,107 @@ class ClassMapperWithEvents : public ClassMapper<T>
     Nidium::Core::Hash<JSEvents *> *m_Events;
 };
 
+// {{{ JSObjectBuilder
+class JSObjectBuilder
```
Comment (paraboul) [inline src/Binding/ClassMapper.h]: This doesn't belong here. Maybe JSUtils

File: src/Binding/JSDebugger.cpp [hunk 2]
```diff
@@ -16,7 +16,7 @@
 namespace Nidium {
 namespace Binding {
 
-// {{{ Preamble
+// {{{ Implementation
 static JSClass DebuggerContext_class = { "DebuggerContext",
```
Comment (paraboul) [inline src/Binding/JSDebugger.cpp]: Looks like it's not finished.

**Comment by efyx** [comment]
I'am picking up this PR to finish the changes.

**Pull request merged**